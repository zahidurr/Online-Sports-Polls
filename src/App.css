.app {
  background: #383737;
  padding: 50px;
}


.column, .column-team-name {
  float: left;
  padding: 0 10px;
  text-align: center;
  font-size: 20px;
}

.row {
  margin: 0 -5px;
}

.card {
  background: #f1f1f1;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  display: flex;
  justify-content: center;
  flex-direction: row;
}

.vote-btn {
  border: none;
  outline: 0;
  display: inline-block;
  padding: 8px;
  color: white;
  background-color: #ce342f;
  text-align: center;
  cursor: pointer;
  width: 120px;
  font-size: 18px;
}

.menu-btn {
  border: none;
  outline: 0;
  display: inline-block;
  padding: 8px;
  color: white;
  background-color: #0366d6;
  text-align: center;
  cursor: pointer;
  width: 120px;
  font-size: 18px;
}

/* Signin input */
input[type=text] {
  width: 80%;
  padding: 13px;
  box-sizing: border-box;
  margin-left: auto;
  margin-right: auto;
}

.team-logo {
  font-size: 250%;
  border: 3px solid rgb(124, 50, 50);
  margin-top: 15px;
  margin-bottom: 15px;
  padding-top: 10px;
  padding-bottom: 10px;
  border-radius: 25px;
}

.play-status {
  font-size: 100%;
  border: 1px solid rgb(0, 0, 0);
  margin: 40px 15px 0 15px;
}

/* Small screen CSS max 600px */
@media only screen and (max-width: 600px) {
  .app {
    background: #383737;
    padding: 10px;
  }

  .column-team-name {
    float: none;
    text-align: center;
  }

  .vote-btn {
    border: none;
    outline: 0;
    display: inline-block;
    padding: 8px 2px 8px 2px;
    color: white;
    background-color: #ce342f;
    text-align: center;
    cursor: pointer;
    width: 70px;
    font-size: 11px;
  }

  .play-status {
    font-size: 80%;
    border: 1px solid rgb(0, 0, 0);
    margin: 40px 5px 0 5px;
  }
}