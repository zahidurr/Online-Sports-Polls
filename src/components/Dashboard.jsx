import React, { useState, useEffect } from "react";
import SigninForm from "./SigninForm";
import ShowName from "./ShowName";
import SignOut from "./SignOut";
import Teams from "./Teams";
import Polls from "./Polls";
import data_file from "../db/test-assignment";

function Dashboard() {
    const initialName = () => String(window.localStorage.getItem('name') || "");
    const [name, setName] = useState(initialName);

    const addName = text => {
        const newName = [ text ];
        setName(newName);
      };

      useEffect(() => {
          window.localStorage.setItem('name', name);

          if(!window.localStorage.getItem("name")) {
            window.localStorage.setItem("teams", JSON.stringify(data_file));
          }
      }, [name]);

    if (name !== "") {
        return ( 
            <React.Fragment>
                <SignOut />
                <ShowName name={name} />
                <Teams />
                <Polls />
            </React.Fragment>
        );
    }
    return ( 
        <SigninForm addName={addName} />
    );
}

  export default Dashboard