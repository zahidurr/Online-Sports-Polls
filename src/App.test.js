import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import App from './App';
import Dashboard from './components/Dashboard';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// Jest Snapshot Testing
describe('App', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Signin - Dashboard', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Dashboard />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

