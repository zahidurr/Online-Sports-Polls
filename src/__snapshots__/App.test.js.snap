// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="app"
>
  <div
    style={
      Object {
        "color": "white",
        "textAlign": "center",
      }
    }
  >
    <h1>
      Glomo - Sports Poll
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        className="input"
        onChange={[Function]}
        placeholder="Enter name"
        required={true}
        type="text"
        value=""
      />
      <button
        className="menu-btn"
        style={
          Object {
            "margin": "10px 0 0 10px",
          }
        }
        type="submit"
      >
         Start
      </button>
    </form>
  </div>
</div>
`;

exports[`Poll - Dashboard snapshot renders 1`] = `
<div
  style={
    Object {
      "color": "white",
      "textAlign": "center",
    }
  }
>
  <h1>
    Glomo - Sports Poll
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="Enter name"
      required={true}
      type="text"
      value=""
    />
    <button
      className="menu-btn"
      style={
        Object {
          "margin": "10px 0 0 10px",
        }
      }
      type="submit"
    >
       Start
    </button>
  </form>
</div>
`;

exports[`Signin - Dashboard snapshot renders 1`] = `
<div
  style={
    Object {
      "color": "white",
      "textAlign": "center",
    }
  }
>
  <h1>
    Glomo - Sports Poll
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="Enter name"
      required={true}
      type="text"
      value=""
    />
    <button
      className="menu-btn"
      style={
        Object {
          "margin": "10px 0 0 10px",
        }
      }
      type="submit"
    >
       Start
    </button>
  </form>
</div>
`;
